### PR TITLE
Chatfeature

### PR DESCRIPTION
**Pull request by TrevorFlanigan**
# Changes
Added Chat.tsx component. Added functionality for backend of chat feature. 
The interface is currently only in the console, with a small input box to add messages to. No frontend yet...
# Tickets
#51

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+# misc
+**.DS_Store
+**.env
+**.env.local
+**.env.development.local
+**.env.test.local
+**.env.production.local
+**.idea
+
+**npm-debug.log*
```
Comment (cowmanjoe) [inline .gitignore]: What are the changes here? Make sure that the line endings aren't changing (e.g. from `\n` to `\r\n`), this is a setting in Git.

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+node_js:
+  - node
+cache:
+  directories:
+    - frontend/node_modules
+    - backend/node_modules
+install:
+  - . scripts/install.sh
+script:
+  - . scripts/build.sh
```
Comment (cowmanjoe) [inline .travis.yml]: Same comment as before, there  appear to be no real changes here.

File: .gitignore (hunk 1 above)
Comment (TrevorFlanigan) [reply .gitignore]: Thanks - changing the setting helped

File: .gitignore [hunk 3]
```diff
@@ -18,4 +18,4 @@
 **.env.production.local
 **.idea
 
-**npm-debug.log*
+**npm-debug.log*
```
Comment (cowmanjoe) [inline .gitignore]: Hmm now it seems like you're getting rid of new lines at the end of some files. If this is unintentional can you put them back?

File: backend/.eslintrc.js [hunk 4]
```diff
@@ -1,17 +1,27 @@
...
+    ecmaFeatures: {
+      jsx: true,
+    },
   },
   rules: {
     "no-var": "error",
     "@typescript-eslint/no-explicit-any": "off",
-    "@typescript-eslint/no-unused-vars": "off"
+    "@typescript-eslint/no-unused-vars": "off",
+    "@typescript-eslint/explicit-function-return-type": "off"
```
Comment (cowmanjoe) [inline backend/.eslintrc.js]: Why turn this off? I personally would prefer to have this feature on (though I'm not sure it is by default)

Comment (cowmanjoe) [reply backend/.eslintrc.js]: Oh I see it was probably included in react/recommended. If this caused errors then probably ok to just leave it off for now and we can turn it on later if we want to

File: frontend/src/components/Chat.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+//   socket.emit('join', data, () => {
+//     alert("printing again");
+//   });
+
+//   return () => {
+//     socket.emit('disconnect');
+
+//     socket.off('disconnect');
+//   }
+// }, [ENDPOINT, window.location.search]);
```
Comment (cowmanjoe) [inline frontend/src/components/Chat.tsx]: Should probably just remove commented code

File: backend/.eslintrc.js [hunk 6]
```diff
@@ -1,17 +1,27 @@
 module.exports = {
   parser: "@typescript-eslint/parser",
   extends: [
+    "plugin:react/recommended",
```
Comment (srijonsaha) [inline backend/.eslintrc.js]: Why are we adding react linting to our backend?

File: frontend/src/components/Chat.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+const display = {
+  messages: []
+};
+
+type MessageProps = {
+  message: string;
+};
+
+// eslint-disable-next-line react/prop-types
```
Comment (srijonsaha) [inline frontend/src/components/Chat.tsx]: If you declare the type in the function parameter you won't need the disable the rule.

File: backend/.eslintrc.js (hunk 6 above)
Comment (TrevorFlanigan) [reply backend/.eslintrc.js]: I'm not sure why that got added, I don't remember intentionally changing the eslint files at all.

File: backend/src/sockets/handler.ts [hunk 8]
```diff
@@ -7,6 +7,11 @@ import Video from "../models/video";
 import uniqid from "uniqid";
 import Database from "../core/database";
 
+interface Message {
```
Comment (cowmanjoe) [inline backend/src/sockets/handler.ts]: Maybe move this to an external file in the `models` folder. I anticipate we will be moving this `models` folder to a shared module

File: frontend/src/components/Room.tsx [hunk 9]
```diff
@@ -16,11 +17,18 @@ interface Props {
   match: any;
 }
 
+interface Message {
```
Comment (cowmanjoe) [inline frontend/src/components/Room.tsx]: Same comment as above but in `frontend/src`

**Pull request merged**